src/cljs/main/broadfcui/page/library/library_page.cljs

```diff
@@ -18,141 +18,144 @@
+                    ;; 112% makes this the same size as "Data Library" / "Workspaces" / "Method Repository" above
+                    [:span {:style {:fontWeight 700 :color (:text-light style/colors) :marginRight "0.5rem"}}
+                     "Matching Cohorts"]
+                    [:span {:style {:fontSize "80%"}}
+                     (let [total (or (:total @state) 0)]
+                       (str total
+                            " Dataset"
+                            (when-not (= 1 total) "s")
+                            " found"))]]
+                   flex/spring]
+           :style {:alignItems "flex-start" :marginBottom 7} ;; 7 makes some lines line up
+           :column-edit-button {:style {:order 1 :marginRight nil}
+                                :anchor :right}}}]))
+    :execute-search
+    (fn [{:keys [refs]} reset-sort?]
+      (let [query-params (merge {:page-number 1} (when reset-sort? {:sort-column nil :sort-order nil}))]
+        (when-not ((@refs "table") :update-query-params query-params)
+          ((@refs "table") :refresh-rows))))
+    :-get-link-props
+    (fn [_ data]
+      (if (= (:workspaceAccess data) "NO ACCESS")
+        {:onClick
+         (fn [_]
+           (comps/push-message
+            {:header "Request Access"
+             :message
+             (if (= (config/tcga-namespace) (:namespace data))
+               [:span {}
+                [:p {} "For access to TCGA protected data please apply for access via dbGaP [instructions can be found "
+                 [:a {:href "https://wiki.nci.nih.gov/display/TCGA/Application+Process"
+                      :target "_blank"}
+                  "here"] "]."]
+                [:p {} "After dbGaP approves your application please link your eRA Commons ID in your FireCloud profile page."]]
+               [:span {}
+                "Please contact "
+                [:a {:target "_blank"
+                     :href (str "mailto:" (:library:contactEmail data))}
+                 (str (:library:datasetCustodian data) " <" (:library:contactEmail data) ">")]
+                " and request access for the "
+                (:namespace data) "/" (:name data) " workspace."])}))}
+        {:href (nav/get-link :workspace-summary
+                             (common/row->workspace-id data))}))
+    :build-aggregate-fields
+    (fn [{:keys [props]}]
+      (reduce
+       (fn [results field] (assoc results field (if (contains? (:expanded-aggregates props) field) 0 5)))
+       {}
+       (:aggregate-fields props)))
+    :pagination
+    (fn [{:keys [this state props locals]}]
+      (fn [{:keys [query-params on-done]}]
+        (let [{:keys [page-number rows-per-page sort-column sort-order]} query-params]
+          (when-not (empty? (:aggregate-fields props))
+            (endpoints/call-ajax-orch
+             (let [from (* (- page-number 1) rows-per-page)
+                   update-aggregates? (or (= 1 page-number) (:initial-render? @locals))]
```

Comment:
you changed the logic from `if (page-number == 1)` to `if (page-number == 1 || initial-render?)`, which is good ... I still think it would be safer to include a check on the value itself:
```
if (page-number == 1 || initial-render? || current-state-value-of-aggregates == nil)
```
to handle the cases we haven't thought of yet (cases we hadn't thought of yet is how we got this bug in the first place)